### ISSUE TITLE
E0747 is not detected if order of arguments is wrong 

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#8306]
## Environment

* **IntelliJ Rust plugin version:** 0.4.163.4319-213-nightly
* **Rust toolchain version:** 1.59.0-nightly (0b42deacc 2021-12-09) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 (CL-213.5744.254)
* **Operating system:** macOS 11.5.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
Error is not detected if order of arguments is wrong 

## Steps to reproduce
There is a piece of code
```rust
mod foo {
    pub struct X<'a, 'b, 'c, T> {
        a: &'a str,
        b: &'b str,
        c: &'c str,
        t: T,
    }
}

fn bar<'a, 'b, 'c, T>(x: foo::X<'a, T, 'b, 'c>) {}
```
## Expected Results
E0747 is detected

## Actual Results
E0747 is not detected.

> error[E0747]: type provided when a lifetime was expected
>   --> src/main.rs:10:37
>    |
> 10 | fn bar<'a, 'b, 'c, T>(x: foo::X<'a, T, 'b, 'c>) {}
>    |       

                          

Related to #8144
